src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGooglePubSubDAO.scala

```diff
@@ -0,0 +1,133 @@
+
+  val pubSubScopes = Seq(PubsubScopes.PUBSUB)
+
+  val httpTransport = GoogleNetHttpTransport.newTrustedTransport
+  val jsonFactory = JacksonFactory.getDefaultInstance
+
+  val serviceAccountClientId: String = clientSecrets.getDetails.get("client_email").toString
+
+  sealed trait AckStatus
+
+  case object MessageAcknowledged extends AckStatus
+  case object MessageNotAcknowledged extends AckStatus
+
+  sealed trait HandledStatus
+
+  case object MessageHandled extends HandledStatus
+  case object MessageNotHandled extends HandledStatus
+  case object NoMessage extends HandledStatus
+
+  def createTopic(topicName: String) = {
+    retry(when500)(() => Future { blocking {
+      executeGoogleRequest(getPubSubDirectory.projects().topics().create(topicToFullPath(topicName), new Topic()))
+    }})
+  }
+
+  def deleteTopic(topicName: String) = {
+    retry(when500)(() => Future { blocking {
+      executeGoogleRequest(getPubSubDirectory.projects().topics().delete(topicToFullPath(topicName)))
+    }})
+  }
+
+  def createSubscription(topicName: String, subscriptionName: String) = {
+    retry(when500)(() => Future { blocking {
+      val subscription = new Subscription().setTopic(topicToFullPath(topicName))
+      executeGoogleRequest(getPubSubDirectory.projects().subscriptions().create(subscriptionToFullPath(subscriptionName), subscription))
+    }})
+  }
+
+  def deleteSubscription(subscriptionName: String) = {
+    retry(when500)(() => Future { blocking {
+      executeGoogleRequest(getPubSubDirectory.projects().subscriptions().delete(subscriptionToFullPath(subscriptionName)))
+    }})
+  }
+
+  def publishMessages(topicName: String, messages: Seq[String]) = {
+    retry(when500)(() => Future { blocking {
+      val pubsubMessages = messages.map(text => new PubsubMessage().encodeData(text.getBytes("UTF-8")))
+      val pubsubRequest = new PublishRequest().setMessages(pubsubMessages)
+      executeGoogleRequest(getPubSubDirectory.projects().topics().publish(topicToFullPath(topicName), pubsubRequest))
+    }})
+  }
+
+  def acknowledgeMessages(subscriptionName: String, ackIds: Seq[String]) = {
+    retry(when500)(() => Future { blocking {
+      val ackRequest = new AcknowledgeRequest().setAckIds(ackIds)
+      executeGoogleRequest(getPubSubDirectory.projects().subscriptions().acknowledge(subscriptionToFullPath(subscriptionName), ackRequest))
+    }})
+  }
+
+  def pullMessages(subscriptionName: String, maxMessages: Int): Future[Seq[ReceivedMessage]] = {
+    retry(when500)(() => Future { blocking {
+      val pullRequest = new PullRequest().setReturnImmediately(true).setMaxMessages(maxMessages) //won't keep the connection open if there's no msgs available
+      val messages = executeGoogleRequest(getPubSubDirectory.projects().subscriptions().pull(subscriptionToFullPath(subscriptionName), pullRequest)).getReceivedMessages
+      if(messages == null)
+        Seq.empty
```

Comment:
This is a nice value-add by your abstraction layer—improving the usability by returning an empty sequence instead of null.